components/pin/hall_effect_sensor/definition.json

```diff
@@ -0,0 +1,14 @@
+{
+    "displayName":"Hall Effect Sensor",
+    "mode":"DIGITAL",
+    "direction":"INPUT",
+    "defaultPeriod":60,
+    "selectPullUp": true,
+    "visualization": {
+      "type": "switch",
+      "offLabel": "Magnet Sensed!",
```

Comment:
FYI: "Magnet Sensed!" will not fit on the switch. Won't break, but will be ugly.

When the user customization is ready you will be able to rapidly test such things before writing the JSON.

Reply:
@lorennorman Ok, what's the maximum # of chars that will fit on the switch?

Reply:
@lorennorman  bumping above for clarification ^

Reply:
Oh hey so sorry I left this hanging, missed a notification!

There's no answer, letters are variable width. What you want to do is open a component viz form on the live site and play with the label inputs with the live preview until you like what you see, then put that here.